comp/core/tagger/taggerimpl/collectors/workloadmeta_main.go

```diff
@@ -103,30 +103,39 @@ func (c *WorkloadMetaCollector) collectStaticGlobalTags(ctx context.Context) {
 		// to read it, for the instances where we are running in an environment where hostname cannot be detected.
 		if cluster := clustername.GetClusterNameTagValue(ctx, ""); cluster != "" {
 			if c.staticTags == nil {
-				c.staticTags = make(map[string]string, 1)
+				c.staticTags = make(map[string][]string, 1)
 			}
-			c.staticTags[clusterTagNamePrefix] = cluster
+			if _, exists := c.staticTags[clusterTagNamePrefix]; !exists {
+				c.staticTags[clusterTagNamePrefix] = []string{}
+			}
+			c.staticTags[clusterTagNamePrefix] = append(c.staticTags[clusterTagNamePrefix], cluster)
 		}
 	}
-	if len(c.staticTags) > 0 {
-		tags := taglist.NewTagList()
+	// These are the global tags that should only be applied to the internal global entity
+	// Whereas the static tags are applied to containers and pods directly as well.
+	globalEnvTags := util.GetGlobalEnvTags()
```

Comment:
Need to create a separate list of the global tags. This is because `c.staticTags` get applied directly to all of the containers and pods and we only want these global tags applied to the `internal://global-entity-id`. There will be some values duplicated in `c.staticTags` and `globalEnvTags`, however, the `tagList` removes duplicates and everything ends up being properly set on the `GlobalEntity`